### PR TITLE
Add RemoveOldestFromCache and save span cache on exit

### PR DESCRIPTION
**Pull request by 0xKiwi**
**What type of PR is this?**
Improvement

**What does this PR do? Why is it needed?**
This PR makes sure that all spans are saved upon closing the slasher, and prunes the cache a bit smarter for historical detection.

### CODE REVIEW

**Comment by terencechain** [comment]
e2e test failed @0xKiwi

File: slasher/db/kv/kv.go [hunk 1]
```diff
@@ -31,12 +33,20 @@ type Config struct {
...
 // Close closes the underlying boltdb database.
 func (db *Store) Close() error {
+	db.spanCache.Purge()
 	return db.db.Close()
 }
 
+// RemoveOldestFromCache clears the oldest key out of the cache obly if the cache is at max capacity.
+func (db *Store) RemoveOldestFromCache(ctx context.Context) uint64 {
+	ctx, span := trace.StartSpan(ctx, "slasherDB.removeOldestFromCache")
+	defer span.End()
```
Comment (rauljordan) [inline slasher/db/kv/kv.go:43]: This span doesn't do anything, as it will run before the return, see https://play.golang.org/p/m-wXudF_BT-

Comment (0xKiwi) [reply slasher/db/kv/kv.go:43]: I see! Thank you for the tip, I'll fix it.

File: slasher/db/kv/kv.go [hunk 2]
```diff
@@ -31,12 +33,21 @@ type Config struct {
 
 // Close closes the underlying boltdb database.
 func (db *Store) Close() error {
+	db.spanCache.Purge()
 	return db.db.Close()
 }
 
+// RemoveOldestFromCache clears the oldest key out of the cache obly if the cache is at max capacity.
```
Comment (rauljordan) [inline slasher/db/kv/kv.go:40]:
```suggestion
// RemoveOldestFromCache clears the oldest key out of the cache only if the cache is at max capacity.
```

**Pull request merged**